src/workflows/airqo_etl_utils/storage/cloud_storage.py

```
@@ -0,0 +1,448 @@
+                blob.upload_from_string(
+                    dataframe.to_json(orient="records"), "application/json"
+                )
+            elif format == "parquet":
+                # For parquet, we need a file-like object or bytes
+                with NamedTemporaryFile() as temp:
+                    dataframe.to_parquet(temp.name)
+                    blob.upload_from_filename(
+                        temp.name, content_type="application/octet-stream"
+                    )
+            else:
+                raise ValueError(f"Unsupported format: {format}")
```

Ruff (0.15.0)

[warning] Abstract `raise` to an inner function

(TRY301)